packages/shorebird_cli/test/src/executables/gradlew_test.dart

```diff
@@ -329,5 +329,135 @@ Make sure you have run "flutter build apk" at least once.''',
         },
       );
     });
+
+    group('exists', () {
+      late Directory tempDir;
+      setUp(() {
+        tempDir = setUpAppTempDir();
+      });
+
+      group('when gradlew does not exist', () {
+        test('returns false', () {
+          expect(gradlew.exists(tempDir.path), isFalse);
+        });
+      });
+
+      group('when gradlew exists', () {
+        group(
+          'when on linux',
+          () {
+            setUp(() {
+              when(() => platform.isLinux).thenReturn(true);
+              when(() => platform.isMacOS).thenReturn(false);
+              when(() => platform.isWindows).thenReturn(false);
+              File(
+                p.join(tempDir.path, 'android', 'gradlew'),
+              ).createSync(recursive: true);
+            });
+
+            test('returns true', () {
+              expect(
+                runWithOverrides(() => gradlew.exists(tempDir.path)),
+                isTrue,
+              );
+            });
+          },
```

Comment:
I think we can use `testOn` for these instead of mocking `platform.isOS`, now that we're running tests on all supported OSes.

Reply:
Also, are the macos tests at all different from the linux tests? Can we have a single test for both?

Reply:
Changed to use `testOn` and kept a single test for macos and linux